### PR TITLE
Add Performance config to resolve bug

### LINKED ISSUE #2349: MySQL performance data stream is missing host configuration details
The MySQL performance data stream is missing configuration parameters.

Discuss: https://discuss.elastic.co/t/mysql-integration-does-not-ingest-performance-metrics/291003

`performance` data stream
```
metricsets: ["performance"]
```

`status` data stream
```
metricsets: ["status"]
hosts:
{{#each hosts}}
  - {{this}}
{{/each}}
{{#if password}}
password: {{password}}
{{/if}}
period: {{period}}
{{#if raw}}
raw: {{raw}}
{{/if}}
{{#if username}}
username: {{username}}
{{/if}}
```

Screenshot 2021-12-06 at 11-46-56 Edit integration - linux-nginx-mysql - Agent policies - Fleet - Elastic

### LINKED ISSUE #3514: MYSQL: performance data_stream configs not passed
Configs are not passed down in the MYSQL data_stream configs.

Also performance data_stream is disabled by default. It seems to be breaking when enabled.

### PR DESCRIPTION
**Pull request by ritalwar**
- Bug

## What does this PR do?

This PR resolves the bug caused by performance data stream in mysql as configs were not getting passed.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

Run the mysql instance, can also use docker container for setup
Add Mysql integration
Enable performance metrics
Data should start flowing through ES.

- Closes #3514
- Closes #2349

### CODE REVIEW

File: packages/mysql/data_stream/performance/fields/fields.yml [hunk 1]
```diff
@@ -9,13 +9,13 @@
...
         - name: last.seen
-          type: date
+          type: keyword
           description: Time at which the digest was most recently seen
         - name: quantile.95
           type: long
           description: The 95th percentile of the statement latency, in picoseconds
         - name: digest
-          type: text
+          type: flattened
```
Comment (ishleenk17) [inline packages/mysql/data_stream/performance/fields/fields.yml:18]: Why is this change needed ?

File: packages/mysql/data_stream/performance/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+description: Pipeline for parsing MySQL performance metrics
+processors:
+- gsub:
+    field: mysql.performance.events_statements.last.seen
+    pattern: ' '
+    replacement: "T"
```
Comment (ishleenk17) [inline packages/mysql/data_stream/performance/elasticsearch/ingest_pipeline/default.yml:7]: Will this conversion work for all the date formats supported by mysql

File: packages/mysql/data_stream/performance/fields/fields.yml (hunk 1 above)
Comment (ritalwar) [reply packages/mysql/data_stream/performance/fields/fields.yml:18]: Using text here was giving parsing exception, then I referred to other integration for similar field i.e https://github.com/elastic/integrations/blob/ebe61d7fd734a1d54064ee123b75567e2a0fa639/packages/aws/data_stream/cloudtrail/fields/fields.yml#L164 and it worked well, so I used this type.
Any other approach suggested ?

File: packages/mysql/data_stream/performance/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (ritalwar) [reply packages/mysql/data_stream/performance/elasticsearch/ingest_pipeline/default.yml:7]: This conversion here is specifically for last.seen field of performance data_stream which is of type timestamp(6) in mysql.
The only format of timestamp supported in mysql is YYYY-MM-DD HH:MM:SS(and YYYY-MM-DD hh:mm:ss[.fraction] with fractional part included ) which holds the combination of date and time, so this conversion should work fine for the data of this field.

Also, as the timestamp field format of mysql contains date and time which is not separated by T,  therefore it was giving date parsing exception in ES and this conversion is inserting that T in place of space which is the format supported by ES, so will not get any parsing exception.

File: packages/mysql/_dev/deploy/docker/Dockerfile [hunk 3]
```diff
@@ -8,4 +8,6 @@ HEALTHCHECK --interval=1s --retries=90 CMD /healthcheck.sh
 COPY healthcheck.sh /healthcheck.sh
 COPY entrypoint.sh /test-entrypoint.sh
 
+RUN chmod -R 777 ./test-entrypoint.sh
```
Comment (ManojS-shetty) [inline packages/mysql/_dev/deploy/docker/Dockerfile:11]: Can we have here permission mode as 644 as a best practice.?

**Review by ManojS-shetty — APPROVED** [review]
